src/App.jsx

```diff
@@ -14,45 +16,83 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+function getSortedGoods(goods, { appliedSort, reverseOrder }) {
+  const visibleGoods = [...goods];
+
+  if (appliedSort) {
+    visibleGoods.sort((good1, good2) => {
+      switch (appliedSort) {
+        case 'alphabetically':
+          return good1.localeCompare(good2);
+
+        case 'length':
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverseOrder) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
+export const App = () => {
+  const [appliedSort, setAppliedSort] = useState('');
+  const [reverseOrder, setReverseOrder] = useState(false);
+
+  const reset = () => {
+    setAppliedSort('');
+    setReverseOrder(false);
+  };
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const visibleGoods = getSortedGoods(goodsFromServer,
+    { appliedSort, reverseOrder });
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={`button is-info ${appliedSort === 'alphabetically' ? '' : 'is-light'}`}
+          onClick={() => setAppliedSort('alphabetically')}
```

Comment:
let's move 'alphabetically' and 'length' to constants (or 1 object with such fields) and use them everywhere to avoid possible errors caused by typo, etc